src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/AtomicPropertyFetchAnalyzer.php

```diff
@@ -205,6 +185,32 @@ public static function analyze(
                     $stmt,
                     new Type\Union($case_values)
                 );
+            } elseif ($prop_name === 'name') {
+                if ($lhs_type_part instanceof Type\Atomic\TEnumCase) {
+                    $statements_analyzer->node_data->setType(
+                        $stmt,
+                        new Type\Union([new Type\Atomic\TLiteralString($lhs_type_part->case_name)])
```

Comment:
`Type::getString($lhs_type_part->case_name)` would have been shorter

Reply:
oh, yeah, I forgot this helper